From d5503b57e841f1473afaa52cb894cd51218a9a5e Mon Sep 17 00:00:00 2001
From: Soul Trace <S-trace@list.ru>
Date: Thu, 23 Jul 2020 15:27:14 +0300
Subject: [PATCH 2/2] sdm660-common: Dnable a2dp offload by default on clover

On my clover and WH-1000M3 I have troubles with this - all supported by the
headset codecs (SBC, AAC, aptX, aptX-HD, LDAC) works fine without A2DP offload,
but with A2DP offload only SBC works correctly, other codecs (AAC, aptX,
aptX-HD gives no audio, and LDAC crashes bluetooth on CrDroid 6.8
and crashes bluetooth and kernel on LineageOS 17.1).

I tried to set persist.vendor.qcom.bluetooth.a2dp_offload_cap=sbc to allow
only SBC offload, but this prop seems to be ignored.

After I set persist.bluetooth.a2dp_offload.disabled=true and rebooted my
clover - A2DP started to work fine with WH-1000M3 headset.
---
 common_prop.mk | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/common_prop.mk b/common_prop.mk
index 28117fbf..7eb3fe8a 100644
--- a/common_prop.mk
+++ b/common_prop.mk
@@ -121,12 +121,21 @@ PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
     persist.vendor.bt.a2dp.aac_whitelist=false \
     persist.vendor.qcom.bluetooth.enable.splita2dp=true \
     persist.vendor.qcom.bluetooth.a2dp_offload_cap=sbc-aptx-aptxhd-aac-ldac \
-    persist.bluetooth.a2dp_offload.disabled=false \
     ro.bluetooth.library_name=libbluetooth_qti.so \
     ro.bluetooth.a2dp_offload.supported=true \
     vendor.audio.feature.a2dp_offload.enable=true \
     vendor.bluetooth.soc=cherokee
 
+ifeq ($(filter clover,$(TARGET_DEVICE)),)
+# A2DP offload works only with SBC on clover and fails or crashes on AAC, aptX, aptX-HD and LDAC
+PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
+    persist.bluetooth.a2dp_offload.disabled=true
+else
+PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
+    persist.bluetooth.a2dp_offload.disabled=false
+endif
+
+
 # Camera
 PRODUCT_PROPERTY_OVERRIDES += \
     persist.camera.privapp.list=org.codeaurora.snapcam \
-- 
2.25.1

